src/types/platforms-types/multi-track-multiprint-contract.ts

```diff
@@ -83,13 +77,13 @@ const mapArtistProfile = ({ apiTrack, nft, contract }: { apiTrack: any, nft?: NF
   }
 };
 
-const mapNFTtoTrackID = (nft: NFT): string => {
-  const trackName = getTrait(nft, 'Track');
-  return `ethereum/${formatAddress(nft.contractAddress)}/${trackName}`;
+const mapNFTtoTrackID = (nft: NFT, extractor?: TitleExtractor): string => {
+  const trackName = extractor ? extractor(nft) : String(nft.tokenId);
+  return `ethereum/${formatAddress(nft.contractAddress)}/${slugify(trackName)}`;
 };
```

Comment:
i don't think we should fallback to tokenId here - given the intention of this mapper is to support contracts that print multiple nfts (ie, multiple token ids) per track, using a tokenId here is likely to cause duplicate tracks in future.

i'd say we should just throw an error if it's not provided.

let's also work to make this more robust&generic and so call id an IdExtractor rather than assuming it's always driven by the title

```suggestion
const mapNFTtoTrackID = (nft: NFT, idExtractor?: IdExtractor): string => {
  if (!idExtractor) {
    throw new Error('No idExtractor provided');
  }
  const id = idExtractor(nft);
  if(!id) {
    throw new Error('Id not extracted correctly');
  }
   return `ethereum/${formatAddress(nft.contractAddress)}/${id}`;
};
```

(the slugify should go in the idExtractor i think, since it's up to the extractor to do?)

Reply:
another thing to think about is how the idExtractor related to titleExtractor. I imagine it could make sense for the default idExtractor to be one that runs the titleExtractor and then slugifies the title.

Though this would mean we need the idExtractor() function can take in the titleExtractor override as a paramater, so the idExtractor knows which titleExtractor to use.

I think this ties in with your comment of "If that's the correct assumption, then passing the NFT along with the function that operates on it to get its title is slightly better than locking down all the NFTs to a single nftFactory" - maybe for each nft we should provide all the overrides it has on a per-nft level (including overrides that may be inherited from their parent contract or parent factory) so that both the mapNFTsToTrackIds and mapNFTtoTrackID and all extractor functions all get the override metadata and so are all flexible to use it however they want.

Do you think this makes sense?